src/single.ts

```diff
@@ -416,15 +422,18 @@ export class RegionRatelimit extends Ratelimit<RegionContext> {
       },
       async getRemaining(ctx: RegionContext, identifier: string) {
 
-        const remainingTokens = await safeEval(
+        const [remainingTokens, refilledAt] = await safeEval(
           ctx,
           tokenBucketRemainingTokensScript,
           "getRemainingHash",
           [identifier],
           [maxTokens],
-        ) as number;
+        ) as [number, number];
 
-        return remainingTokens;
+        return {
+          remaining: remainingTokens,
+          reset: refilledAt === -1 ? Date.now() + intervalDuration : refilledAt + intervalDuration
```

Comment:
`-1` Let's replace this magic number with constant var.

Also, this would make it more redable:
```
const start = Date.now() + intervalDuration // start and end just for example, we can find more reader friendly name
const end= refilledAt + intervalDuration
reset: refilledAt === -1 ? Date.now() + + intervalDuration
```

Reply:
updated the code